flash/data/base_viz.py

```diff
@@ -1,101 +1,139 @@
 
 
-class BaseViz(FlashCallback):
+class BaseViz(BaseDataFetcher):
     """
-    This class is used to profile ``Preprocess`` hook outputs and visualize the data transformations.
-    It is disabled by default.
+    This Base Class is used to create visualization tool on top of :class:`~flash.data.process.Preprocess` hooks.
 
-    batches: Dict = {"train": {"to_tensor_transform": [], ...}, ...}
+    Override any of the ``show_{preprocess_hook_name}`` to receive the associated data and visualize them.
 
-    """
+    Example::
+
+        from flash.vision import ImageClassificationData
+        from flash.data.base_viz import BaseViz
+
+        class CustomBaseViz(BaseViz):
+
+            def show_load_sample(self, samples: List[Any], running_stage):
+                # plot samples
+
+            def show_pre_tensor_transform(self, samples: List[Any], running_stage):
+                # plot samples
+
+            def show_to_tensor_transform(self, samples: List[Any], running_stage):
+                # plot samples
+
+            def show_post_tensor_transform(self, samples: List[Any], running_stage):
+                # plot samples
 
-    def __init__(self, enabled: bool = False):
-        self.batches = {k: {} for k in _STAGES_PREFIX.values()}
-        self.enabled = enabled
-        self._preprocess = None
+            def show_collate(self, batch: List[Any], running_stage):
+                # plot batch
 
-    def _store(self, data: Any, fn_name: str, running_stage: RunningStage) -> None:
-        if self.enabled:
-            store = self.batches[_STAGES_PREFIX[running_stage]]
-            store.setdefault(fn_name, [])
-            store[fn_name].append(data)
+            def show_per_batch_transform(self, batch: List[Any], running_stage):
+                # plot batch
 
-    def on_load_sample(self, sample: Any, running_stage: RunningStage) -> None:
-        self._store(sample, "load_sample", running_stage)
+        class CustomImageClassificationData(ImageClassificationData):
 
-    def on_pre_tensor_transform(self, sample: Any, running_stage: RunningStage) -> None:
-        self._store(sample, "pre_tensor_transform", running_stage)
+            @staticmethod
+            def configure_data_fetcher(*args, **kwargs) -> BaseDataFetcher:
+                return CustomBaseViz(*args, **kwargs)
 
-    def on_to_tensor_transform(self, sample: Any, running_stage: RunningStage) -> None:
-        self._store(sample, "to_tensor_transform", running_stage)
+        dm = CustomImageClassificationData.from_folders(
+            train_folder="./data/train",
+            val_folder="./data/val",
+            test_folder="./data/test",
+            predict_folder="./data/predict")
 
-    def on_post_tensor_transform(self, sample: Tensor, running_stage: RunningStage) -> None:
-        self._store(sample, "post_tensor_transform", running_stage)
+        # visualize a ``train`` batch
+        dm.show_train_batches()
```

Comment:
@tchaton I'm trying to replicate this exact example and I get:
`AttributeError: 'CustomImageClassificationData` object has no attribute 'show_train_batches'

Reply:
You have the `test_base_data_fetcher` if you want to check it out.